### PR TITLE
Add error code to failed graphql response

### PR DESCRIPTION
**Pull request by korycins**

### CODE REVIEW

**Comment by django-queries** [comment]
Here is the report for 686bc5a28661d63862c461687f67e0926e31334a (korycins/saleor @ fix/always_show_error_code_in_graphql)
Base comparison is 13a4654fc922ee1b86b0fe5c9b720b887eaca110.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  add billing address to checkout            	         34	         34	             20
  add shipping to checkout                   	          7	          7	              0
  checkout payment charge                    	         14	         14	              0
  complete checkout                          	          6	          6	              0
  create checkout                            	         48	         48	             24

# api.benchmark homepage
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  retrieve main menu                         	          5	          5	              0
  retrieve product list                      	          4	          4	              0
  retrieve secondary menu                    	          5	          5	              0
  retrieve shop                              	          2	          2	              0

# api.benchmark product
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  product details                            	         15	         15	              3
  retrieve product attributes                	         13	         13	              2

# api.benchmark variant
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  product variant bulk create                	         51	         51	              3
  retrieve variant list                      	         15	         15	              6

# api product sorting attributes
  test name                                  	left count 	right count	duplicate count
  -------------------------------------------	-----------	-----------	---------------
  sort product not having attribute data     	         21	         21	              0
```

</p>
</details>

File: saleor/graphql/views.py [hunk 1]
```diff
@@ -200,6 +200,7 @@ def format_error(error):
 
         logger.error("Exception information:", exc_info=exc)
 
+        result["extensions"] = {"exception": {"code": type(exc).__name__}}
```
[the remark this answers is not in the document]
Comment (korycins) [reply saleor/graphql/views.py]: Do you have a better idea, how to provide a type of error to the dashboard?

[the remark this answers is not in the document]
Comment (maarcingebala) [reply saleor/graphql/views.py]: Frontend needs to somehow determine read-only errors in our public demo. This information will be available here as the status code.

[the remark this answers is not in the document]
Comment (NyanKiyoshi) [reply saleor/graphql/views.py]: Yes, but it gives way too much information to tell what every error happening in backend is. Some error names could be very descriptive and could be used against the backend.

We could have a custom error class, i.e. `PublicAPIException` or anything that has a `verbose_name` or `code` property, then from that we could return a error code e.g. `InternalError` by default, then `ReadOnlyException`, ...

[the remark this answers is not in the document]
Comment (NyanKiyoshi) [reply saleor/graphql/views.py]: One doesn't need to know what exactly went wrong behind the scenes if that was something unexpected.

**Pull request merged**